tools/pre-commit-hook.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
```

Comment:
Ok but this does not solve the issues for macOS and GUI. This will fail in macOS as need to be `/usr/bin/env python3`. And for GUI we are not already in the correct env so ruff and Cython-lint won't be find.

Reply:
Just make it `#!python3`.

I don't know what to do about the GUI: must be an issue with all hooks?

Reply:
ah cool if this is portable, then we should do that anyway 👍 

For the GUI, env issue, other tools create a custom hook with correct paths during the instal process of the hook. Not sure if we can move around a manual action here.

But that would be the only action since there is no	`mapfile` anymore.

Reply:
Easiest may be to give a one liner for finding the current interpreter and put that in the first line of the hook.

Reply:
I found out that `#!python3` is, in fact, invalid.